server/middleware/registration/index.js

```diff
@@ -0,0 +1,125 @@
+const bcrypt = require('bcrypt');
+const {
+  HttpStatusCodes: { StatusCodes },
+  DataException
+} = require('../../utils/constants');
+const { REQUEST_TIMEOUT, OK, BAD_REQUEST, GATEWAY_TIMEOUT } = StatusCodes;
+const mongoUtil = require('../../utils/mongoUtil');
+const JWT = require('jsonwebtoken');
+const { v4 } = require('uuid');
+const registrationLogger = require('./registrationLogger');
+
+const resObj = (statusCode, message, data = {}) => ({
+  statusCode,
+  message,
+  data,
+});
+
+const isFormValid = ({ body }) => {
+  const formFields = [
+    // 'role_id',
+    'firstName',
+    'lastName',
+    'title',
+    'email',
+    'password'
+  ];
+  
+  for (let i = 0; i < formFields.length; i++) {
+    const field = formFields[i];
+    if (!Object.hasOwnProperty.call(body, formFields[i]) || body[field].length === 0) {
+      return false;
+    }
+  }
+
+  const emailPattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  if (!emailPattern.test(body.email)) {
+    return false;
+  } else {
+    return true;
+  }
+
+};
+
+const hashForm = ({ body }) => {
+  const saltRounds = 10;
+  body.email = body.email.toLowerCase();
+  const { password, email, title, role_id, lastName, firstName } = body;
+  const userPayload = { password: bcrypt.hashSync(password, saltRounds), email, title, role_id, lastName, firstName };
```

Comment:
bug fix: we need to save th e role_id as type int32 in the users collection or there will be a type mismatch when it tries to compare the role_id in the common menu collection vs the users collection when fetching the common menu data after registering or signing in.

`const userPayload = {
    password: bcrypt.hashSync(password, saltRounds),
    email,
    title,
    role_id: parseInt(role_id),
    lastName,
    firstName,
  };`